src/main/java/gov/cabinetoffice/gap/adminbackend/services/ApplicationFormSectionService.java

```diff
@@ -90,11 +91,13 @@ public void updateSectionStatus(final Integer applicationId, final String sectio
         this.applicationFormRepository.save(applicationForm);
     }
 
-    public void updateSectionTitle(final Integer applicationId, final String sectionId, final String title) {
+    public void updateSectionTitle(final Integer applicationId, final String sectionId, final String title, final Integer revision) {
 
         ApplicationFormEntity applicationForm = this.applicationFormRepository.findById(applicationId)
                 .orElseThrow(() -> new NotFoundException("Application with id " + applicationId + " does not exist"));
 
+        ApplicationFormUtils.verifyApplicationFormRevision(revision, applicationForm);
```

Comment:
This should be the last thing we do prior to saving

Reply:
Would adding a @Version annotation on the entity help with this?